contextual-identities/context.js

```diff
@@ -0,0 +1,54 @@
+function eventHandler(event) {
+  if (event.target.dataset.action == 'create') {
+    browser.tabs.create({
+      url: 'about:blank',
+      cookieStoreId: event.target.dataset.identity
+    });
+  }
+  if (event.target.dataset.action == 'close-all') {
+    browser.tabs.query({
+      cookieStoreId: event.target.dataset.identity
+    }).then((tabs) => {
+      browser.tabs.remove(tabs.map((i) => i.id));
+    });
+  }
+  event.preventDefault();
+}
+
+function createOptions(node, identity) {
+  for (let option of ['Create', 'Close All']) {
+    let a = document.createElement('a');
+    a.href = '#';
+    a.innerText = option;
+    a.dataset.action = option.toLowerCase().replace(' ', '-');
+    a.dataset.identity = identity.cookieStoreId;
+    a.addEventListener('click', eventHandler);
+    node.appendChild(a);
+  }
+}
+
+var div = document.getElementById('identity-list');
+
+if (browser.contextualIdentities === undefined) {
+  div.innerText = 'browser.contextualIdentities not available, do you have the pref privacy.userContext.enabled set to true?';
```

Comment:
It's unfortunate that if the pref is false and you toggle it, you need to reload the add-on for it to work. I don't see how you can get around that but it might be worth supplementing this message:

'browser.contextualIdentities is not available. Check that the privacy.userContext.enabled pref is set to true, and reload the add-on'

Or something better than that.